projects/ng2-smart-table/src/lib/ng2-smart-table.component.ts

```diff
@@ -108,6 +113,38 @@ export class Ng2SmartTableComponent implements OnChanges {
     this.rowClassFunction = this.grid.getSetting('rowClassFunction');
   }
 
+  selectRow(index: number, switchPageToSelectedRowPage: boolean = this.grid.getSetting('switchPageToSelectedRowPage')): void {
+    if (!this.grid) {
+      return;
+    }
+    const dataAmount: number = this.source?.data?.length;
+    if (index < 0 || (typeof dataAmount === 'number' && index >= dataAmount)) {
+      this.grid.dataSet.deselectAll();
+      return;
+    }
+
+    if (switchPageToSelectedRowPage) {
+      const source: DataSource = this.source;
+      const paging: { page: number, perPage: number } = source.getPaging();
+      const page: number = Math.ceil((index + 1) / paging.perPage);
+
+      if (page !== paging.page) {
+        (this.grid.settings as any).selectedRowIndex = index;
+        source.setPage(page);
+        return;
```

Comment:
Shouldn't we set selection after changing the page?

Reply:
no, because above we have already set the required index

Reply:
Oh, right 🙂
Why don't we change `grid.settings.selectedRowIndex` in all other cases when the page doesn't change?